modules/home-environment.nix

```diff
@@ -256,7 +256,7 @@ in
 
     home.sessionVariables = mkOption {
       default = {};
-      type = types.attrs;
+      type = types.lazyAttrsOf types.anything;
```

Comment:
types.anything ? not str/int ?  seems like types.attrs didn't enforce anything either so that's probably fine.

Reply:
Has `types.raw` or `types.unconditional` been merged? I believe they are substitutes for `types.anything`

Reply:
`unconditional` has to do with moving the `lazy` to the element type instead of the attribute type, it's mostly an implementation detail.

`raw` is for when absolutely no processing is wanted. I think this would disable `mkOverride`, and would also not use `mergeEqualOption` so that multiple equal definitions would raise an error, which seems less nice.

But yes we really want a more precise type. Anything that can be `toString`ed was previously accepted, but maybe we should restrict that to `oneOf [ str path int float ]` or something.

Reply:
The next step would be to make that the `freeformType` of a submodule, which is the core of https://github.com/nix-community/home-manager/pull/1769, so that more precise options (e.g. concatenating lists with `:` for PATH) can be added later.

Reply:
sounds awesome.